ee/control/consumers/remoterestartconsumer/remoterestartconsumer.go

```diff
@@ -79,7 +80,7 @@ func (r *RemoteRestartConsumer) Do(data io.Reader) error {
 
 	// Perform the restart by signaling actor shutdown, but delay slightly to give
 	// the actionqueue a chance to process all actions and store their statuses.
-	go func() {
+	gowrapper.Go(context.TODO(), r.slogger, func() {
```

Comment:
I love wrapping all our goroutines to help us be aware of panics! I think maybe we could improve the ergonomics of the wrapping so that we don't to pass an empty func if we just want the default onPanic behavior.

Some suggestions in no particular order:

* Having another gowrapper func that doesn't require an OnPanic func.
* Updating gowrapper.Go to do an nil check on the func so we could just pass nil as the 4th param.
* Make the gowrapper into a struct and use an options pattern.

These are just suggestions, I'de also accept the PR as is.

Reply:
I like the first option especially, feels really easy to slot in with our existing usage. Maybe `Go(ctx context.Context, slogger *slog.Logger, goroutine func()` (for usage that doesn't require an OnPanic func) and `GoWithRecoveryAction(ctx context.Context, slogger *slog.Logger, goroutine func(), onPanic func(r any))` (for usage that does require an OnPanic func). I'd be fine with that in a follow-up PR as well.

Reply:
Sounds good, I'll merge this one and start with the follow up on a seperate PR